backend/tools/keycloak/sync/Factory.cs

```diff
@@ -0,0 +1,386 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Pims.Tools.Keycloak.Sync.Exceptions;
+using Pims.Tools.Keycloak.Sync.Extensions;
+using Pims.Tools.Keycloak.Sync.Models;
+using KModel = Pims.Tools.Keycloak.Sync.Models.Keycloak;
+
+namespace Pims.Tools.Keycloak.Sync
+{
+    /// <summary>
+    /// Factory class, provides a way to sync Keycloak roles, groups and users with PIMS.
+    /// </summary>
+    public class Factory : IFactory
```

Comment:
Could the naming here be more informative?

Reply:
Yes but I'm not going to change it for a job that runs once and may never be run again.